erizo/src/test/rtp/RtpAudioMuteHandlerTest.cpp

```diff
@@ -0,0 +1,221 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <queue>
+
+#include <rtp/RtpAudioMuteHandler.h>
+#include <rtp/RtpHeaders.h>
+#include <MediaDefinitions.h>
+#include <WebRtcConnection.h>
+
+#include <vector>
+
+static constexpr uint16_t kVideoSsrc = 1;
+static constexpr uint16_t kAudioSsrc = 2;
+static constexpr uint16_t kArbitrarySeqNumber = 12;
+static constexpr uint16_t kFirstSequenceNumber = 0;
+static constexpr uint16_t kLastSequenceNumber = 65535;
+
+using ::testing::_;
+using ::testing::IsNull;
+using ::testing::Args;
+using ::testing::Return;
+using erizo::dataPacket;
+using erizo::packetType;
+using erizo::AUDIO_PACKET;
+using erizo::VIDEO_PACKET;
+using erizo::IceConfig;
+using erizo::RtpMap;
+using erizo::RtpAudioMuteHandler;
+using erizo::WebRtcConnection;
+using erizo::Pipeline;
+using erizo::InboundHandler;
+using erizo::OutboundHandler;
+using erizo::Worker;
+using std::queue;
+
+class MockWebRtcConnection: public WebRtcConnection {
+ public:
+  MockWebRtcConnection(std::shared_ptr<Worker> worker, const IceConfig &ice_config,
+                       const std::vector<RtpMap> rtp_mappings) :
+    WebRtcConnection(worker, "", ice_config, rtp_mappings, nullptr) {}
+
+  virtual ~MockWebRtcConnection() {
+  }
+};
+
+class Reader : public InboundHandler {
+ public:
+  MOCK_METHOD2(read, void(Context*, std::shared_ptr<dataPacket>));
+};
+
+class Writer : public OutboundHandler {
+ public:
+  MOCK_METHOD2(write, void(Context*, std::shared_ptr<dataPacket>));
+};
+
+class RtpAudioMuteHandlerTest : public ::testing::Test {
+ public:
+  RtpAudioMuteHandlerTest() : ice_config() {}
+
+ protected:
+  virtual void SetUp() {
+    scheduler = std::make_shared<Scheduler>(1);
+    worker = std::make_shared<Worker>(scheduler);
+    connection = std::make_shared<MockWebRtcConnection>(worker, ice_config, rtp_maps);
+
+    connection->setVideoSinkSSRC(kVideoSsrc);
+    connection->setAudioSinkSSRC(kAudioSsrc);
+
+    pipeline = Pipeline::create();
+    audio_mute_handler = std::make_shared<RtpAudioMuteHandler>(connection.get());
+    reader = std::make_shared<Reader>();
+    writer = std::make_shared<Writer>();
+
+    pipeline->addBack(writer);
+    pipeline->addBack(audio_mute_handler);
+    pipeline->addBack(reader);
+    pipeline->finalize();
+  }
+
+  virtual void TearDown() {
+  }
+
+  std::shared_ptr<dataPacket> createDataPacket(uint16_t seq_number, packetType type) {
+    erizo::RtpHeader *header = new erizo::RtpHeader();
+    header->setSeqNumber(seq_number);
+
+    if (type == AUDIO_PACKET) {
+      header->setSSRC(kAudioSsrc);
+    } else {
+      header->setSSRC(kVideoSsrc);
+    }
+
+    return std::make_shared<dataPacket>(0, reinterpret_cast<char*>(header), sizeof(erizo::RtpHeader), type);
+  }
+
+  std::shared_ptr<dataPacket> createNack(uint16_t seq_number, packetType type, int additional_packets = 0) {
```

Comment:
maybe this is the right PR to create a common helper for all these methods (createData, createNack, Matchers, etc.) and to share them with other UTs. Feel free to add it here, or we can do it in a separate PR if you prefer.